WalletWasabi/Crypto/ZeroKnowledge/NonInteractive/FiatShamirTransform/Statement.cs

```diff
@@ -0,0 +1,71 @@
+using System.Linq;
+using System.Collections.Generic;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge.NonInteractive.FiatShamirTransform
+{
+	public class Statement : IStatement
+	{
+		public Statement(LinearRelation.Statement statement)
+		{
+			LinearRelation = statement;
+		}
+
+		// Although in principle this could be an interface type, since only linear
+		// relations are needed for our proofs this is specialized to that type.
+		public LinearRelation.Statement LinearRelation { get; }
+
+		// Plug witness data into this statement. The enumerator should have exactly
+		// one vector, corresponding to the sizes of the equations in the base case
+		// of a linear relation and disjunction, and in the case of conjunction
+		// since we use conjunctive normal form, provide exactly one witness for
+		// each sub-statement.
+		public IKnowledge ToKnowledge(IEnumerable<ScalarVector> witnesses)
+		{
+			Guard.NotNullOrEmpty(nameof(witnesses), witnesses);
+			Guard.True(nameof(witnesses), witnesses.Count() == 1);
```

Comment:
This is good to assert what you want. I would use:

```suggestion
			Guard.Same(nameof(witnesses), witnesses.Count(), 1);
```

because it gives you useful exception message saying what you have passed to the function. (Typically it's zero or 
too much".)